### PR TITLE
Refactor permission trees

### PR DESCRIPTION
**Pull request by tdraier**
fixes: [#1358](https://github.com/dust-tt/tasks/issues/1358)

## Description

This refactor the PermissionTree in a more reusable and configurable way. 
Main changes are : 
- Handling of selection is now fully controlled, when passing a treeSelectionModel / setTreeSelectionModel object. No more internal states duplicating external state
- One single ContentNodeTree which takes a simple hook as parameter, used to fetch the nodes under a specific parent. Same tree can be used for datasources, views, or anything that list content nodes
- All logic regarding permissions is moved outside
- Handle the partial checkbox selection when a subnode is selected, and the disabled checkboxes under a selected parent

## Risk

Can break trees in different places, need to be tested on frontedge first. Usages are : 
- Connection management , readonly (notion, ..) and not readonly (google, slack,..)
- Assistant details view
- Slack integration (channels selection)
- Poke data source view

## Deploy Plan

deploy front

### CODE REVIEW

File: front/components/DataSourceViewPermissionTree.tsx
Comment (tdraier) [reply front/components/DataSourceViewPermissionTree.tsx:41]: Actually using useCallback (even if it would be ok) breaks the rules-of-hook (React Hook "usePokeConnectorPermissions" cannot be called inside a callback. React Hooks must be called in a React function component or a custom React Hook function.). 
Defining a dynamic hook a in render function is not a violation of any hook rule so should not be blocked by a linter ?

File: front/components/ContentNodeTree.tsx
Comment (flvndvd) [inline front/components/ContentNodeTree.tsx:156]: Might worth a small comment to explain that `setSelectedNodes` is not set while selection is first loaded.

**Review by flvndvd — APPROVED** [review]
LGTM 🙏 If you can add a comment and move some nested reduce to a function to help understand

**Pull request merged**
